File: backend/src/main/kotlin/com/casadosreclamos/dto/ClientDto.kt
package com.casadosreclamos.dto

import com.casadosreclamos.model.Client
import java.util.stream.Collectors

data class ClientDto(
    var id: Long?,
    var banner: String?,
    var name: String?,
    var fiscalNumber: String?,
    var email: String?,
    var phone: String?,
    var address: String?,
    var postalCode: String?,
    var city: String?,
    var country: String?,
    var note: String?,
    var images: List<ImageDto>?
) {
    constructor(client: Client) : this(
        client.id,
        client.banner.name,
        client.name,
        client.fiscalNumber,
        client.email,
        client.phone,
        client.address,
        client.postalCode,
        client.city,
        client.country,
        client.note,
        client.images.stream().map { ImageDto(it) }.collect(Collectors.toList())
    )
}

File: frontend/src/models/ClientDto.ts
import ImageDto from "@models/ImageDto";

export default class ClientDto {
  id: number | string;
  banner: string | null = null;
  name: string;
  address: string;
  postalCode: string;
  fiscalNumber: string;
  email: string;
  phone: string;
  city: string;
  country: string;
  images: ImageDto[] = [];

  constructor(obj?: Partial<ClientDto>) {
    if (obj) {
      this.id = obj.id;
      this.banner = obj.banner;
      this.name = obj.name;
      this.fiscalNumber = obj.fiscalNumber;
      this.email = obj.email;
      this.phone = obj.phone;
      this.address = obj.address;
      this.postalCode = obj.postalCode;
      this.city = obj.city;
      this.country = obj.country;
      this.images = obj.images.map((i) => new ImageDto(i));
    }
  }

  clear() {
    this.id = "";
    this.banner = null;
    this.name = "";
    this.fiscalNumber = "";
    this.email = "";
    this.phone = "";
    this.address = "";
    this.postalCode = "";
    this.city = "";
    this.country = "";
    this.images = [];
  }
}
